src/List/__docs__/list.md

```diff
@@ -0,0 +1,46 @@
+# List
+
+The `List` component maps a collection of data objects into an array of elements. It also manages the selection and focus of those elements.
+
+## Usage
+
+```jsx
+import { List } from '@magento/peregrine';
+
+const basicItems = new Map()
+    .set('s', 'Small')
+    .set('m', 'Medium')
+    .set('l', 'Large')
+
+<List
+    classes={{ root: 'foo' }}
+    items={basicItems}
+    render="select"
+    renderItem="option"
+/>
+
+const complexItems = new Map()
+    .set('s', { id: 's', value: 'Small' })
+    .set('m', { id: 'm', value: 'Medium' })
+    .set('l', { id: 'l', value: 'Large' })
+
+// let `Select` and `Option` be React components
+<List
+    classes={{ root: 'bar' }}
+    items={complexItems}
+    render={Select}
+    renderItem={Option}
+    onSelectionChange={selection => { console.log(selection); }}
+/>
+```
+
+## Props
+
+Prop Name | Required? | Description
+--------- | :-------: | :----------
+classes | ❌ | A classname hash
+items | ✅ | A keyed collection of data objects, preferably an ES2015 `Map`
```

Comment:
The proptype for this component is explicitly requiring only a `Map` right now:

![image](https://user-images.githubusercontent.com/5233399/39531182-3e3dd1f8-4df0-11e8-85f8-6abd65af0660.png)